qualang_tools/utils/qua_utils.py

```diff
@@ -0,0 +1,237 @@
+import numpy as np
+
+
+def qua_list(var, array, var_type):
+    """
+    Function parametrizing the QUA `for_` loop from a python array.
+    :param var: the QUA variable that will be looped over (int or fixed).
+    :param array: a Python list or numpy array containing the values over which `a` will be looping. The spacing must be constant or logarithmic and it must be a Python array.
+    :param var_type: type of the a variable, can be either 'int' or 'fixed'.
+    :return: QUA for_ loop parameters (var, init, cond, update) as defined in https://qm-docs.qualang.io/api_references/qua/dsl_main?highlight=for_#qm.qua._dsl.for_.
+    """
+    assert len(array) > 0, "The array must be of length >0"
+
+    if np.isclose(np.std(np.diff(array)), 0):
+        increment = "lin"
+    elif np.isclose(np.std(array[1:] / array[:-1]), 0):
+        increment = "log"
+    else:
+        raise ValueError(
+            "The spacing of the input array must be linear or logarithmic."

```

Comment:
I would add: Please use with `for_each_()` for scanning arbitrary values

Reply:
Good point, done!